robot_smach_states/src/robot_smach_states/util/designators/navigation/waypoint_constraints.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import absolute_import
+
+# System
+import inspect
+import pprint
+
+# ROS
+import rospy
+from geometry_msgs.msg import *
+
+# TU/e Robotics
+from robot_skills.util.entity import Entity
+from ..core import Designator
+from ..checks import check_resolve_type
+from cb_planner_msgs_srvs.msg import OrientationConstraint, PositionConstraint
+
+
+class WayPointConstaintsDesignator(Designator):
+    def __init__(self, robot, waypoint_designator, radius=0.15, look_at_designator=None, name=None):
+        super(WayPointConstaintsDesignator, self).__init__(resolve_type=tuple, name=name)
+        self.robot = robot
+
+        check_resolve_type(waypoint_designator, Entity)  # Check that the waypoint_designator resolves to an Entity
```

Comment:
Repeating in comment what the code says already is useless at best (and worse if code and comment disagree).

Let the code say what is being called or computed at its line, use comment for higher level notions that the code doesn't say.

Examples of higher level notions are for example the aim of the next block of code, or why this step must be done here (and not elsewhere), or why the seemingly weird piece of code is actually correct.